spec/components/pretty_text_spec.rb

```diff
@@ -1465,6 +1465,14 @@ def expect_cooked_match(raw, expected_cooked)
       expect(PrettyText.cook("f.o")).to match_html("<p>test</p>")
     end
 
+    it "does not replace hashtags and mentions" do
+      Fabricate(:user, username: "test")
+      Fabricate(:category, slug: "test")
+      Fabricate(:watched_word, action: WatchedWord.actions[:replace], word: "test", replacement: "discourse")
+
+      expect(PrettyText.cook("@test #test test").scan('discourse').size).to eq(1)
```

Comment:
I think we need to test for the actual content here. If there is a bug that accidentally replaces `@test` with `@discourse` I think this assertion will still pass.

Reply:
That's not true because the output will be `@discourse #test discourse` and `discourse` will be twice in the output.